exa/core/data.py

```diff
@@ -95,19 +93,16 @@ def _observe_source(self, change):
             self.call_args = []
             self.call_kws = {}
 
-    @validate('name')
-    def _validate_name(self, prop):
-        name = prop['value']
-        self.log.debug(f"lowercasing {name}")
-        return name.lower()
-
-    @default('name')
-    def _default_name(self):
-        return self.__class__.__name__
-
     @validate('cardinal')
     def _validate_cardinal(self, prop):
-        c = prop['value']
+        """The cardinal concept refers to a "global" index
+        shared across multiple data objects. Perhaps its
+        API should live in and be controlled by the Container?
+        Here can serve as a pass-through to parent if it exists.
+        Should it need to be present in data?
+        """
```

Comment:
I think this is a really good idea. When a data object is attached to a container, does the data object have a reference to its container?